packages/discovery-react-components/src/utils/document/documentUtils.ts

```diff
@@ -208,6 +204,86 @@ export function uniqRects(rects: DOMRectList): Partial<DOMRectList> {
   );
 }
 
+/**
+ * Finds the descendant node that contains the given offset.
+ * This is done efficiently via a binary search.
+ * @param parentNode node to search the descendants of
+ * @param offset value to find within the nodes
+ * @return the descendant node, or null if none is found
+ */
+function findContainingNodeWithin(parentNode: HTMLElement, offset: number) {
```

Comment:
Just a note: it might be nice to create some basic unit tests for these utils. Not required -- not sure how time consuming writing test cases would be for these.

Reply:
Yeah, that probably makes sense. :+1: